### PR TITLE
added dto, controller, mapper

### PR DESCRIPTION
**Pull request by mmosii**

### CODE REVIEW

File: src/main/java/mmosii/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/books")
```
Comment (don-bigdad) [inline src/main/java/mmosii/bookstore/controller/BookController.java:17]: in task(Book Endpoints:
GET: /api/books (Retrieve book catalog))

**Review by don-bigdad — APPROVED** [review]

File: src/main/java/mmosii/bookstore/config/MapperConfig.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import org.mapstruct.InjectionStrategy;
+import org.mapstruct.NullValueCheckStrategy;
+
+@org.mapstruct.MapperConfig(
+        componentModel = "spring",
+        injectionStrategy = InjectionStrategy.CONSTRUCTOR,
+        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
+        implementationPackage = "<PACKAGE_NAME>.impl"
+)
+public class MapperConfig {
```
Comment (NazarSmith228) [inline src/main/java/mmosii/bookstore/config/MapperConfig.java]: it is better to specify @MapperConfig directly on a mapper class

File: src/main/java/mmosii/bookstore/dto/BookDto.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package mmosii.bookstore.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class BookDto {
```
Comment (NazarSmith228) [inline src/main/java/mmosii/bookstore/dto/BookDto.java:7]: is it possible to use record for DTO?

File: src/main/java/mmosii/bookstore/mapper/BookMapper.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+import mmosii.bookstore.dto.BookDto;
+import mmosii.bookstore.dto.CreateBookRequestDto;
+import mmosii.bookstore.model.Book;
+import org.mapstruct.Mapper;
+
+@Mapper(config = MapperConfig.class)
+public interface BookMapper {
+    BookDto toDto(Book book);
+
+    Book fromDto(CreateBookRequestDto requestDto);
```
Comment (NazarSmith228) [inline src/main/java/mmosii/bookstore/mapper/BookMapper.java]: `toBook` sounds better :)

File: src/main/java/mmosii/bookstore/repository/impl/BookRepositoryImpl.java [hunk 5]
```diff
@@ -41,6 +38,14 @@ public Book save(Book book) {
         }
     }
 
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Book book = session.find(Book.class, id);
+            return Optional.ofNullable(book);
```
Comment (NazarSmith228) [inline src/main/java/mmosii/bookstore/repository/impl/BookRepositoryImpl.java]: you can inline this :)

File: src/main/java/mmosii/bookstore/service/impl/BookServiceImpl.java [hunk 6]
```diff
@@ -1,29 +1,38 @@
...
     }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
-
+    public BookDto getBookById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(()
+                -> new EntityNotFoundException("Can't find book by Id " + id));
+        return bookMapper.toDto(book);
```
Comment (NazarSmith228) [inline src/main/java/mmosii/bookstore/service/impl/BookServiceImpl.java]: you can do this via single Optional methods chain

don't forget that Optional has a `map` method :)

**Review by NazarSmith228 — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution

File: src/main/java/mmosii/bookstore/dto/BookDto.java (hunk 3 above)
Comment (mmosii) [reply src/main/java/mmosii/bookstore/dto/BookDto.java:7]: Application works good with record but have problem with maven checkstyle.  " unexpected token: record "

**Review by liliia-ponomarenko — APPROVED** [review]
Great 😊

**Pull request merged**